python/ray/_private/services.py

```diff
@@ -919,7 +919,23 @@ def start_redis(node_ip_address,
         # If no port is given, fallback to default Redis port for the primary
         # shard.
         if port is None:
-            port = ray_constants.DEFAULT_PORT
+            default_port = ray_constants.DEFAULT_PORT
+            if sys.platform == "win32":
+                port = default_port
+                num_choices = 200
+                a_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
Can we simplify this logic a bit? 

```
for port in range(default_port, default_port + port_range):
  location = (node_ip_address, port)
  try:
    a_socket.bind(location)
  except socket.error as e:
    if e.errno != errno.EADDRINUSE:
      raise
  else:
    break
a_socket.close()
```

Reply:
I can do that. Though then both instances (there can be n in general) of ray will see same port as available all the times since the same for loop will be running everywhere. That's why I pick the next port as some random integer from the port range.